gordon_janitor/main.py

```diff
@@ -180,8 +180,16 @@ def run(config_root):
     loop = asyncio.new_event_loop()
     try:
         loop.run_until_complete(_run(plugins, debug_mode))
-    finally:
+    except Exception as e:
+        logging.error(f'An error occurred during the janitor run: {e}')
         loop.close()
+        raise e
+
+    logging.info('Gordon-janitor run complete.')
```

Comment:
I think there's a chance that we'd never get to this section of code since you `raise` above. I would put this whole chunk under a `finally` clause so that metrics are always sent. As well, the `loop.close` in the `except` clause should be removed since it's in the `finally` bit. Does that make sense?

Reply:
That was intentional. I wanted both the log message and metric to be sent only if Janitor completes normally. The test verifies that this is the case.

Reply:
Oh I see - maybe the metric / log can be more clear that it _successfully_ completed?

I do think we still need a `finally` so we cleanup the loop in any case. With that, we can add an `else`, i.e.:

```py
try:
    loop.run...
except Exception as e:
    logging.error(...)
    raise e
else:  # only happens when `try` is successful
    logging.info(...)
    metrics...
finally:
    loop.close()
    # anything else we always want to happen whether we succeed or fail
```